src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/AttachCaseCallbackService.java

```diff
@@ -388,13 +388,22 @@ private void updateSupplementaryEvidence(
         CaseDetails theCase = ccdApi.getCase(targetCaseCcdRef, callBackEvent.exceptionRecordJurisdiction);
         List<Map<String, Object>> targetCaseDocuments = getScannedDocuments(theCase);
 
-        //This is done so exception record does not change state if there is a document error
-        checkForDuplicatesOrElse(
+        ExceptionRecordAttachDocumentConnectives erDocumentConnectives = calculateDocumentConnectives(
             callBackEvent.exceptionRecordDocuments,
-            targetCaseDocuments,
-            ids -> throwDuplicateError(targetCaseCcdRef, ids)
+            targetCaseDocuments
         );
 
+        if (erDocumentConnectives.hasDuplicates()) {
```

Comment:
question : If documents were already attached should we not proceed with payments ?
Duplicate document scenario would only be possible when callback timed out and caseworker has retried ?

Reply:
i'll handle this scenario in future prs ;). this is refactor so that document connectives can be expanded and recognise such scenario